### ISSUE TITLE
to_cubed method

### ISSUE THREAD

**Issue by TomNicholas** [intake/intake#752]
There's currently a `to_zarr` method, the implementation of which just calls `dask.array.from_zarr()` on a zarr store url. I would like to add a `to_cubed` method which does the same thing using `cubed.from_zarr()`. This would allow intake users to access data as [cubed arrays](https://github.com/tomwhite/cubed).

Where do I need to add this? In the `ZarrSource` class here? Or somewhere in intake-xarray?

Thanks

**Comment by martindurant** [comment]
Yes, you certainly could add a to_cubed, and the right place would probably be in the ZarrSource.

**However**, intake is going through a transition. #737 shows the bones of what will be "intake 2.0", in which the data definition ("zarr type, at this URL") will be separated from readers ("call this function to produce an instance of type X", roughly), so that you can define the set of kwargs for any reader of interest for your zarr in one place, or even multiple sets of kwargs for the same reader. Thus, the different possible outcomes from a given dataset are all enumerated and described together.

**Comment by TomNicholas** [comment]
> the data definition ("zarr type, at this URL") will be separated from readers

That sounds great, and very compatible with [what I added to xarray recently](https://github.com/pydata/xarray/pull/7019), allowing `chunked_array_type` and `from_array_kwargs` to be passed to `open_dataset`.

That refactor looks like a big project though, and as long as you are still releasing intake, adding a `to_cubed` methods seems to me to be a reasonable short-term step.

**Comment by martindurant** [comment]
> That refactor looks like a big project though, and as long as you are still releasing intake, adding a to_cubed methods seems to me to be a reasonable short-term step.

Agreed, I am not opposed :)
